pkg/monitor/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) 2024 Zededa, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# syntax=docker/dockerfile:1
+ARG RUST_VERSION=lfedge/eve-rust:1.80.1
+FROM --platform=$BUILDPLATFORM ${RUST_VERSION} AS toolchain-base
+ARG TARGETARCH
+
+FROM toolchain-base AS target-amd64
+ENV CARGO_BUILD_TARGET="x86_64-unknown-linux-musl"
+
+FROM toolchain-base AS target-arm64
+ENV CARGO_BUILD_TARGET="aarch64-unknown-linux-musl"
+
+FROM toolchain-base AS target-riscv64
+ENV CARGO_BUILD_TARGET="riscv64gc-unknown-linux-gnu"
+
+FROM target-$TARGETARCH AS toolchain
+RUN echo "Cargo target: $CARGO_BUILD_TARGET"
+
+FROM toolchain AS planer
+# get sources from GIT
+ADD --keep-git-dir=true https://github.com/lf-edge/eve-monitor-rs.git /app
+
+WORKDIR /app
+# create a recipe
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM toolchain AS cacher
+# copy the recipe
+WORKDIR /app
+COPY --from=planer /app/recipe.json recipe.json
+# build the dependencies
+RUN cargo chef cook --release --recipe-path recipe.json
+
+# building the final image
+FROM toolchain AS builder
+ADD --keep-git-dir=true https://github.com/lf-edge/eve-monitor-rs.git /app
+WORKDIR /app
+
+# copy prebuilt dependencies
+# and the cargo directory with crates.io index
+COPY --from=cacher /app/target /app/target
+COPY --from=cacher $CARGO_HOME $CARGO_HOME
+RUN echo "Cargo target: $CARGO_BUILD_TARGET"
+
+RUN cargo build --release
+RUN cargo sbom > sbom.spdx.json
+RUN cp /app/target/$CARGO_BUILD_TARGET/release/monitor /app/target/
+
+
+FROM lfedge/eve-alpine:591df01e581889c3027514c8a91feaca1c8ad49f AS runtime
+ENV PKGS="kbd pciutils usbutils"
+RUN eve-alpine-deploy.sh
+
+FROM scratch
+COPY --from=runtime /out/usr/bin/openvt /usr/bin/openvt
+COPY --from=runtime /out/usr/bin/loadkeys /usr/bin/loadkeys
+COPY --from=runtime /out/bin/dmesg /usr/bin/dmesg
+COPY --from=runtime /out/usr/bin/lsusb /usr/bin/lsusb
+COPY --from=runtime /out/usr/bin/lspci /usr/bin/lspci
+COPY --from=runtime /out/usr/share/keymaps/xkb/us.map.gz /usr/share/keymaps/xkb/us.map.gz
+COPY --from=runtime /out/lib /lib
+COPY --from=runtime /out/usr/lib /usr/lib
+
+# copy busybox and sh but just for debugging
+COPY --from=runtime /out/bin/sh /bin/sh
```

Comment:
in this case, `/bin/sh` will be a symbolic link to busybox, right? So instead of copy both sh and busybox (line below), it makes sense to copy only busybox and run `/bin/busybox --install -s <DESTINATION_DIR>` to create all symbolic links of supported commands...

Reply:
@rene your suggestion will not work as-is but I implemented a similar one. check out updated code. pretty interesting fact